### PR TITLE
Parse object type in method argument and return 

### PR DESCRIPTION
**Pull request by mpetrun5**
Linked: #89

### CODE REVIEW

File: packages/schema/parse/src/extract/index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export type SchemaExtractor = (astNode: DocumentNode, typeInfo: TypeInfo) => voi
 
 export const extractors: SchemaExtractor[] = [
   extractUserTypes,
-  extractQueryTypes,
   extractImportedObjectTypes,
+  extractQueryTypes,
```
Comment (mpetrun5) [inline packages/schema/parse/src/extract/index.ts:13]: Is the order of extractors important for something else?
I need extractor to parse importedObjectTypes so I can recognize it is a object in `NamedType` parsing.

File: packages/schema/parse/src/extract/imported-query-types.ts [hunk 2]
```diff
@@ -139,9 +141,18 @@ const visitorEnter = (typeInfo: TypeInfo, state: State) => ({
 
     if (method && argument) {
       // Argument value
-      argument.scalar = createScalarDefinition(
-        argument.name, modifier + node.name.value, state.nonNullType
-      );
+      if (
```
Comment (mpetrun5) [inline packages/schema/parse/src/extract/imported-query-types.ts:144]: Currently, imported query logic and query logic for extracing some types is the same so we should
probably extract that somewhere

File: packages/schema/parse/src/__tests__/cases/sanity/output.ts [hunk 3]
```diff
@@ -86,6 +94,27 @@ export const output: TypeInfo = {
             createScalarPropertyDefinition("arg", "String", true)
           ],
           return: createScalarPropertyDefinition("queryMethod", "Int", true)
+        },
+        {
+          ...createMethodDefinition("query", "importedObjectMethod"),
+          arguments: [
+            createObjectPropertyDefinition(
```
Comment (mpetrun5) [inline packages/schema/parse/src/__tests__/cases/sanity/output.ts:101]: Do I need uri and namespace here for imported objects or I can just treat as regular user object?

**Review by mpetrun5 — COMMENTED** [review]
@dOrgJelli 
General idea for parsing
Left a couple of questions for which I am not sure of

File: packages/schema/parse/src/__tests__/cases/sanity/output.ts (hunk 3 above)
Comment (dOrgJelli) [reply packages/schema/parse/src/__tests__/cases/sanity/output.ts:146]: The UI & namespace are available on the TestImport_Object definition, so no need to carry those properties through to this instance.

Comment (dOrgJelli) [reply packages/schema/parse/src/__tests__/cases/sanity/output.ts:146]: So in short, all import metadata is only needed on the ImortedObjectDefinition, and not on the arguments + return type definitions.

File: packages/schema/parse/src/extract/index.ts (hunk 1 above)
Comment (dOrgJelli) [reply packages/schema/parse/src/extract/index.ts:17]: Ordering should be independent, as the extractors **shouldn't** require TypeInfo data from other extractors. This is of course an ideal scenario, and may break down in practice. Let me know if you there are dependencies between the extractors and we can figure out how to best handle this.

Comment (mpetrun5) [reply packages/schema/parse/src/extract/index.ts:17]: Have dependecy here https://github.com/Web3-API/prototype/blob/fa6e10dad19e8b53b7c90117b3e87c88551aa4a7/packages/schema/parse/src/extract/query-utils.ts#L36

To recognize if it is object or scalar in named type extraction I need to check if the objects is defined somewhere
as I don't have `kind` from node.
Only way I found out was to check if that object exists in already extracted objects in `typeInfo`.

Comment (dOrgJelli) [reply packages/schema/parse/src/extract/index.ts:17]: Chatted about this, will be handled in another PR.

**Pull request merged**